Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -588,26 +588,29 @@ void HUDController::UpdateVisualCooldowns(GameObject* canvas, bool isMain, int s
 void HUDController::OnHealthLost(GameObject* targetCanvas, int health) {
 	bool isFang = (targetCanvas->GetID() == fangHealthMainCanvas->GetID());
 
-	GameObject* obj = targetCanvas->GetChild(health);
+	int prevLivesToUse = isFang ? prevLivesFang : prevLivesOni;
+	std::vector<int>& remainingTimeActiveIndexesToUse = isFang ? remainingTimeActiveIndexesFang : remainingTimeActiveIndexesOni;
+	float* remainingDurableHealthTimesToUse = isFang ? remainingDurableHealthTimesFang : remainingDurableHealthTimesOni;
 
+	int healthLost = prevLivesToUse - health;
+	int healthToUse = health;
 
-	if (isFang) {
-		remainingTimeActiveIndexesFang.push_back(health);
-		remainingDurableHealthTimesFang[health] = timeToFadeDurableHealthFeedback;
-	} else {
-		remainingTimeActiveIndexesOni.push_back(health);
-		remainingDurableHealthTimesOni[health] = timeToFadeDurableHealthFeedback;
-	}
+	for (int i = 0; i < healthLost; i++) {
 
-	if (!obj) return;
+		GameObject* obj = targetCanvas->GetChild(healthToUse);
 
-	HealthLostInstantFeedback* fb = GET_SCRIPT(obj, HealthLostInstantFeedback);
+		remainingTimeActiveIndexesToUse.push_back(healthToUse);
+		remainingDurableHealthTimesToUse[healthToUse] = timeToFadeDurableHealthFeedback;
 
-	if (!fb) return;
+		if (!obj) return;
```

Comment:
return or continue?

Reply:
return, because !obj mean obj==nullptr, which means GET_SCRIPT will generate a crash.

Reply:
Continue in a for halt to next iteration

Reply:
Okay now I get it. Done.